### PR TITLE
application not found return error. fix #685

### LINKED ISSUE #685: [BUG]Should return an error for non-existent applications
**Describe the bug**
application does not exist should return an error instead of an empty info.

**To Reproduce**
```console
vela show notfound
or
vela status notfound

About:

  Name:      	notfound
  Created at:	0001-01-01 00:00:00 +0000 UTC
  Updated at:	0001-01-01 00:00:00 +0000 UTC


Environment:

  Namespace:	default

Services:
```

**Expected behavior**
should be like kubectl
```console
kubectl get pods notfound
Error from server (NotFound): pods "notfound" not found
```

**Screenshots**

**Cluster information**

**Additional context**

### PR DESCRIPTION
**Pull request by majian159**
issues #685

### CODE REVIEW

File: pkg/application/errors.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+import (
+	"errors"
+)
+
+var (
+	ErrNotFound = errors.New("not found")
+)
+
+func IsNotFound(err error) bool {
```
Comment (zzxwill) [inline pkg/application/errors.go:11]: As `IsNotFound ` isn't widely used, I prefer not create `errors.go`.

File: pkg/application/app.go [hunk 2]
```diff
@@ -69,7 +69,15 @@ func Load(envName, appName string) (*Application, error) {
...
+	app, err := LoadFromFile(filepath.Join(appDir, appName+".yaml"))
+
+	if err != nil {
+		if IsNotFound(err) {
+			return nil, fmt.Errorf(`applications "%s" %w`, appName, err)
+		}
+	}
+
+	return app, nil
 }
```
Comment (zzxwill) [inline pkg/application/app.go:81]: Please also write a unit-test for this case. Thanks.

File: pkg/application/app_test.go [hunk 3]
```diff
@@ -136,3 +137,18 @@ services:
 		assert.Equal(t, c.ExpTraits, traits, caseName)
 	}
 }
+
+func TestLoadNotExistsApplication(t *testing.T) {
+	caseName := "load not exists application"
+
+	now := time.Now().Unix()
+	appName := fmt.Sprintf("test-app-%d", now)
```
Comment (zzxwill) [inline pkg/application/app_test.go:145]: Cool.

File: pkg/application/app_test.go [hunk 4]
```diff
@@ -136,3 +137,18 @@ services:
...
+
+	app, err := Load(types.DefaultEnvName, appName)
+
+	assert.Nil(t, app, caseName)
+	assert.Error(t, err, caseName)
+
+	errString := fmt.Sprintf(`applications "%s" not found`, appName)
```
Comment (zzxwill) [inline pkg/application/app_test.go:152]:
```suggestion
	errString := fmt.Sprintf(`application "%s" not found`, appName)
```

**Comment by wonderflow** [comment]
@majian159  Hi, thanks for your contribution, could you fix the CI

**Comment by majian159** [comment]
@wonderflow I can't seem to solve e2e-test.

File: pkg/commands/exec.go [hunk 5]
```diff
@@ -114,11 +115,11 @@ func (o *VelaExecOptions) Init(ctx context.Context, c *cobra.Command, argsIn []s
 	if err != nil {
 		return err
 	}
+	o.Env = env
```
Comment (zzxwill) [inline pkg/commands/exec.go:118]: Why do you move line:121 to L:181. Seems `o.Env = nil` all the time.

File: pkg/commands/exec_test.go [hunk 6]
```diff
@@ -2,22 +2,28 @@ package commands
...
 	"github.com/stretchr/testify/assert"
 	corev1 "k8s.io/api/core/v1"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
 	"k8s.io/client-go/kubernetes/fake"
 	"k8s.io/kubectl/pkg/cmd/exec"
 	cmdtesting "k8s.io/kubectl/pkg/cmd/testing"
 
+	"github.com/oam-dev/kubevela/pkg/oam"
+
```
Comment (zzxwill) [inline pkg/commands/exec_test.go:22]: Please delete this blank line.

File: pkg/commands/exec.go [hunk 7]
```diff
@@ -14,11 +14,12 @@ import (
 	cmdexec "k8s.io/kubectl/pkg/cmd/exec"
 	k8scmdutil "k8s.io/kubectl/pkg/cmd/util"
 
+	"github.com/oam-dev/kubevela/pkg/oam"
+
```
Comment (zzxwill) [inline pkg/commands/exec.go:18]: This blank line should also be deleted.

File: pkg/application/app.go [hunk 8]
```diff
@@ -50,7 +50,7 @@ func LoadFromFile(fileName string) (*Application, error) {
 	_, err = os.Stat(fileName)
 	if err != nil {
 		if os.IsNotExist(err) {
-			return newApplication(nil, tm), nil
+			return nil, err
```
Comment (zzxwill) [inline pkg/application/app.go:53]: Returning `nil`: +1.
Returning `err`: -1 as it happens when the application was created before.

File: pkg/application/app.go [hunk 9]
```diff
@@ -69,7 +69,16 @@ func Load(envName, appName string) (*Application, error) {
...
+	app, err := LoadFromFile(filepath.Join(appDir, appName+".yaml"))
+
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, fmt.Errorf(`application "%s" not found`, appName)
+		}
+		return nil, err
+	}
+
+	return app, nil
```
Comment (zzxwill) [inline pkg/application/app.go:81]: It's better to raise the exception in its callers. When in creating, it should not return any error, when in `vela show`, it should.

File: pkg/commands/exec.go (hunk 5 above)
Comment (majian159) [reply pkg/commands/exec.go:117]: The application's acquisition logic may be wrong, which will affect the env.

File: pkg/commands/portforward_test.go [hunk 10]
```diff
@@ -2,20 +2,23 @@ package commands
...
+
 	"github.com/stretchr/testify/assert"
 	corev1 "k8s.io/api/core/v1"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	k8sfake "k8s.io/client-go/kubernetes/fake"
 	"k8s.io/kubectl/pkg/cmd/portforward"
 	cmdtesting "k8s.io/kubectl/pkg/cmd/testing"
 
+	"github.com/oam-dev/kubevela/pkg/oam"
+
```
Comment (wonderflow) [inline pkg/commands/portforward_test.go:19]: nit

File: pkg/commands/exec_test.go [hunk 11]
```diff
@@ -2,13 +2,18 @@ package commands
 
 import (
 	"context"
+	"fmt"
 	"os"
 	"testing"
 
 	"github.com/spf13/cobra"
+
+	k8scmdutil "k8s.io/kubectl/pkg/cmd/util"
```
Comment (wonderflow) [inline pkg/commands/exec_test.go:11]: nit

**Review by wonderflow — APPROVED** [review]
Good Job! Generally LGTM, leave to @zzxwill to merge

File: pkg/commands/portforward_test.go (hunk 10 above)
Comment (majian159) [reply pkg/commands/portforward_test.go]: removed

File: pkg/commands/exec_test.go (hunk 11 above)
Comment (majian159) [reply pkg/commands/exec_test.go]: removed

**Pull request merged**